ops/terraform/services/server/modules/bfd_server_error_alerts/lambda_src/bfd_server_errors_alerter.py

```diff
@@ -0,0 +1,327 @@
+            logger.info(
+                'Query "%s" has not finished running (current status: %s), waiting 1 second...',
+                start_query_response["queryId"],
+                response["status"],
+            )
+            time.sleep(1)
+            response = logs_client.get_query_results(queryId=start_query_response["queryId"])
+    except logs_client.exceptions.ClientError:
+        logger.error(
+            'Retrieving results for query ID "%s" has failed due to an unrecoverable error:',
+            exc_info=True,
+        )
+        return
+
+    logger.info(
+        "Query (ID: %s) has completed with status %s",
+        start_query_response["queryId"],
+        response["status"],
+    )
+
+    if not response["results"]:
+        logger.info(
+            "No results found. It is likely there have been no errors in the past %s"
+            " second(s). Stopping...",
+            LOG_LOOKBACK_SECONDS,
+        )
+        return
+
+    # Response has the following format:
+    # [
+    #     [
+    #         {"field": "partner", "value": "..."},
+    #         {"field": "op", "value": "..."},
+    #         {"field": "count(status)", "value": "..."},
+    #     ],
+    #     [...],
+    # ]
+    # Parse the response to create a dict of partner to error count with the following format:
+    # {
+    #     "partner1": {
+    #         "op1": ...,
+    #         "op2": ...,
+    #     },
+    #     "...": {...}
+    # }
+    partner_to_errors: dict[str, dict[str, int]] = {}
+    for result_row_list in response["results"]:
+        partner = __get_value_by_key("partner", result_row_list)
+        op_to_count = {
+            __get_value_by_key("op", result_row_list): int(
+                __get_value_by_key("count(status)", result_row_list)
+            )
+        }
+        partner_to_errors[partner] = (
+            partner_to_errors[partner] | op_to_count
+            if partner in partner_to_errors
+            else op_to_count
+        )
+
+    total_errors = sum(sum(x.values()) for x in partner_to_errors.values())
+    logger.info(
+        "Per-partner error results: %s, total errors: %s in past %s second(s)",
+        partner_to_errors,
+        total_errors,
+        LOG_LOOKBACK_SECONDS,
+    )
+
+    per_partner_tables: dict[str, str] = {}
+    for partner, op_counts in partner_to_errors.items():
+        longest_op_chars_count = max(len(x) for x in op_counts.keys())
+        longest_err_chars_count = max(len(str(x)) for x in op_counts.values())
+        table_block_str = (
+            f"```\n{'Operation':<{longest_op_chars_count}} {'Error Count':<{longest_err_chars_count}}\n"
+        )
+        for op, count in op_counts.items():
+            table_block_str += (
+                f"{op:<{longest_op_chars_count}} {count:<{longest_err_chars_count}}\n"
+            )
+        table_block_str += "```"
+
+        per_partner_tables[partner] = table_block_str
```

Comment:
Slack webhooks [do not seem to support Markdown-formatted tables](https://stackoverflow.com/questions/37080635/slack-webhook-html-table), and using the rudimentary [`fields`](https://stackoverflow.com/a/55734396) section for display a two-column table is limited to 10 entries.

So, this code generates an appropriately formatted "console table", using code fences to display unformatted, monospaced text. For example:

```
Operation                  Error Count
this/is/a/long/operation() 99999999
short/operation()          10000000
...
```